stargate-redis/app/com/salesforce/mce/stargate/services/impl/SessionTrackingServiceRedisImpl.scala

```diff
@@ -51,13 +51,7 @@ class SessionTrackingServiceRedisImpl @Inject() (config: Configuration)(implicit
     logger.info(s"""Deleting key "$key" in redis.""")
     val status = jedisCluster.del(key)
     logger.info(s"""Status reply when deleting key "$key": $status""")
-    if (status == 1) {
-      logger.info(s"""Successfully destroyed session for key "$key".""")
-      Future.successful(true)
-    } else {
-      logger.error(s"""Failed to destroy session for key "$key". Invalid status "$status" returned.""")
-      Future.successful(false)
-    }
+    Future.successful(status > 0)
```

Comment:
Refering to bug https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000007sCTFIA2/view

> Current Impl: If the response is 1 then it is success else it if failed.
> Expected: Redis returns the number of key affected, so 0 is success as no key exists in redis.

Should this line be `Future.successful(status >= 0)`?

Reply:
So the question is, what's treating `Future.successful(false)` as failures?

Reply:
yes. That was baffling me too. The current code is marking the future as successful, the false or true is just the accompanying value.

Reply:
That was my preference also. Discussed this with Kexin and he suggested we return a bool that indicate if cache is cleared or not

Reply:
Currently the logout code in McSsoController is what is calling destroy. It is not using any return values.